src/OrbitGl/TimeGraph.cpp

```diff
@@ -727,33 +712,26 @@ void TimeGraph::UpdateChildrenPosAndContainerSize() {
   // After we set positions.
   float timegraph_current_x = GetPos()[0];
   float timegraph_current_y = GetPos()[1];
+  const float total_right_margin = layout_.GetRightMargin() + vertical_slider_->GetWidth();
 
+  timeline_ui_->SetWidth(GetWidth() - total_right_margin);
   timeline_ui_->SetPos(timegraph_current_x, timegraph_current_y);
-  timegraph_current_y += timeline_ui_->GetHeight();
 
+  // TODO(b/230441392): Margin between timeline and tracks shouldn't be part of the timeline.
+  timegraph_current_y += timeline_ui_->GetHeight();
+  track_container_->SetWidth(GetWidth() - total_right_margin);
   track_container_->SetPos(timegraph_current_x, timegraph_current_y);
-  timegraph_current_y += track_container_->GetHeight();
-
-  slider_->SetPos(timegraph_current_x, timegraph_current_y);
 
   vertical_slider_->SetWidth(layout_.GetSliderWidth());
-  vertical_slider_->SetPos(GetWidth() - vertical_slider_->GetWidth(),
-                           track_container_->GetPos()[1]);
-
-  // TODO(b/230442062): Refactor this to be part of Slider::UpdateLayout(). Set visibility of the
-  // vertical slider should be inside of this method.
-  UpdateVerticalSliderFromWorld();
+  vertical_slider_->SetPos(GetWidth() - vertical_slider_->GetWidth(), timegraph_current_y);
 
-  // Now we can set width of every child.
-  const float vertical_slider_width =
-      (vertical_slider_->GetVisible() ? vertical_slider_->GetWidth() : 0.f);
-  const float total_right_margin = layout_.GetRightMargin() + vertical_slider_width;
-  timeline_ui_->SetWidth(GetWidth() - total_right_margin);
-  track_container_->SetWidth(GetWidth() - total_right_margin);
-  slider_->SetWidth(GetWidth() - vertical_slider_width);
+  timegraph_current_y += track_container_->GetHeight();
+  slider_->SetWidth(GetWidth() - vertical_slider_->GetWidth());
```

Comment:
Nit: Shall we consider renaming this as `horizontal_slider_`?

Reply:
I made this last Friday :) https://github.com/google/orbit/pull/3652. Now that I see I will have conflicts